modules/gapi/doc/00-root.markdown

```diff
@@ -0,0 +1,143 @@
+
+    - @subpage gapi_core
+
+      Core G-API operations - arithmetic, boolean, and other matrix
+      operations;
+
+    - @subpage gapi_imgproc
+
+      Image processing functions: color space conversions, various
+      filters, etc.
+
+# API Example {#gapi_example}
+
+A very basic example of G-API pipeline is shown below:
+
+@code{.cpp}
+    #include <opencv2/opencv.hpp>
+    #include <opencv2/gapi.hpp>
+    #include <opencv2/gapi/core.hpp>
+    #include <opencv2/gapi/imgproc.hpp>
+
+    int main(int argc, char *argv[])
+    {
+        cv::VideoCapture cap;
+        if (argc > 1) cap.open(argv[1]);
+        else cap.open(0);
+        CV_Assert(cap.isOpened());
+
+        cv::GMat in;
+        cv::GMat vga      = cv::gapi::resize(in, cv::Size(), 0.5, 0.5);
+        cv::GMat gray     = cv::gapi::BGR2Gray(vga);
+        cv::GMat blurred  = cv::gapi::blur(gray, cv::Size(5,5));
+        cv::GMat edges    = cv::gapi::canny(blurred, 32, 128, 3);
+        cv::GMat b,g,r;
+        std::tie(b,g,r)   = cv::gapi::split3(vga);
+        cv::GMat out      = cv::gapi::merge3(b, g | edges, r);
+        cv::GComputation ac(in, out);
+
+        cv::Mat input_frame;
+        cv::Mat output_frame;
+        CV_Assert(cap.read(input_frame));
+        do
+        {
+            ac.apply(input_frame, output_frame);
+            cv::imshow("output", output_frame);
+        } while (cap.read(input_frame) && cv::waitKey(30) < 0);
+
+        return 0;
+    }
+@endcode
+
+<!-- TODO align this code with text using marks and itemized list -->
+
+G-API is a separate OpenCV module so its header files have to be
+included explicitly. The first four lines of `main()` create and
+initialize OpenCV's standard video capture object, which fetches
+video frames from either an attached camera or a specified file.
+
+G-API pipelie is constructed next. In fact, it is a series of G-API
+operation calls on cv::GMat data. The important aspect of G-API is
+that this code block is just a declaration of actions, but not the
+actions themselves. No processing happens at this point, G-API only
+tracks which operations form pipeline and how it is connected. G-API
+_Data objects_ (here it is cv::GMat) are used to connect operations
+each other. `in` is an _empty_ cv::GMat signalling that it is a
+beginning of computation.
+
+After G-API code is written, it is captured into a call graph with
+instantiation of cv::GComputation object. This object takes
+input/output data references (in this example, `in` and `out`
+cv::GMat objects, respectively) as parameters and reconstructs the
+call graph based on all the data flow between `in` and `out`.
+
+cv::GComputation is a thin object in sense that it just captures which
+operations form up a computation. However, it can be used to execute
+computations -- in the following processing loop, every captured frame (a
+cv::Mat `input_frame`) is passed to cv::GComputation::apply().
+
+![Example pipeline running on sample video 'vtest.avi'](pics/demo.png)
```

Comment:
Please use .jpg format (with ~90% quality) for real-like images.
PNG is not very good for this purpose.

Other images (schemes) are fine to store in png format.

Reply:
Okay